Component configuration guides (#41)
kinvolk/lokomotive: docs/configuration-guides/components/flatcar-linux-update-operator.md

# Flatcar Container Linux Update Operator Configuration Reference for Lokomotive

## Contents

* [Introduction](#introduction)
* [Prerequisites](#prerequisites)
* [Configuration](#configuration)
* [Argument Reference](#argument-reference)
* [Installation](#installation)
* [Uninstallation](#uninstallation)

## Introduction

> This component is for a node reboot controller for Kubernetes running Flatcar Container Linux

Reviewer: ```suggestion
This component is a controller that manages node reboots for nodes running Flatcar Container Linux
```

images. When a reboot is needed after updating the system via
[update_engine](https://github.com/coreos/update_engine), the operator will drain the node before
rebooting it.

## Prerequisites

* A Lokomotive cluster accessible via `kubectl`.

## Configuration

This component does not require any specific configuration.

> Empty configuration block is also accepted as valid configuration.

Reviewer: ```suggestion
An empty configuration block is also accepted as valid configuration.
```

Flatcar Container Linux update operator component configuration example:

```hcl
component "flatcar-linux-update-operator" {}
```

> In some cases, you would want to prevent a certain node from rebooting from the operator. To do

Reviewer: ```suggestion
In some cases, you would want to prevent a certain node from rebooting by the operator. To do
```

that:

```bash
kubectl label nodes NODENAME flatcar-linux-update.v1.flatcar-linux.net/reboot-pause=true
```

> For more details visit the [GitHub](https://github.com/kinvolk/flatcar-linux-update-operator)

Reviewer: ```suggestion
For more details visit the [Flatcar Container Linux update operator GitHub repository](https://github.com/kinvolk/flatcar-linux-update-operator).
```

> repository.

Reviewer: ```suggestion
```
